cirq-core/cirq/testing/consistent_decomposition.py

```diff
@@ -47,3 +47,20 @@ def assert_decompose_is_consistent_with_unitary(val: Any, ignoring_global_phase:
     else:
         # coverage: ignore
         np.testing.assert_allclose(actual, expected, atol=1e-8)
+
+
+def assert_decompose_ends_at_default_gateset(val: Any):
+    """Ensures that all cirq gate decompositions end at the default cirq gateset."""
+
+    # pylint: disable=unused-variable
+    __tracebackhide__ = True
+    # pylint: enable=unused-variable
+    if protocols.is_parameterized(val):
+        return
+    args = () if isinstance(val, ops.Operation) else (tuple(devices.LineQid.for_gate(val)),)
+    dec_once = protocols.decompose_once(val, None, *args)
+    if dec_once is None:
+        # _decompose_ is NotImplemented, so silently return.
+        return
```

Comment:
Same question as above, if not implemented wouldn't we want to fail loudly ?

Reply:
Addressed the feedback, see my comment above.